torch/nn/utils/spectral_norm.py

```diff
@@ -14,32 +14,34 @@ def __init__(self, name='weight', n_power_iterations=1, eps=1e-12):
         self.eps = eps
 
     def compute_weight(self, module):
-        weight = module._parameters[self.name + '_org']
-        u = module._buffers[self.name + '_u']
+        weight = getattr(module, self.name + '_org')
+        u = getattr(module, self.name + '_u')
         height = weight.size(0)
         weight_mat = weight.view(height, -1)
-        for _ in range(self.n_power_iterations):
-            # Spectral norm of weight equals to `u^T W v`, where `u` and `v`
-            # are the first left and right singular vectors.
-            # This power iteration produces approximations of `u` and `v`.
-            v = normalize(torch.matmul(weight_mat.t(), u), dim=0, eps=self.eps)
-            u = normalize(torch.matmul(weight_mat, v), dim=0, eps=self.eps)
-
-        sigma = torch.dot(u, torch.matmul(weight_mat, v))
-        weight.data /= sigma
+        with torch.no_grad():
+            for _ in range(self.n_power_iterations):
+                # Spectral norm of weight equals to `u^T W v`, where `u` and `v`
+                # are the first left and right singular vectors.
+                # This power iteration produces approximations of `u` and `v`.
+                v = normalize(torch.matmul(weight_mat.t(), u), dim=0, eps=self.eps)
+                u = normalize(torch.matmul(weight_mat, v), dim=0, eps=self.eps)
+
+                sigma = torch.dot(u, torch.matmul(weight_mat, v))
```

Comment:
Do we need to compute the sigma in the loop? It's different from before

Reply:
No, good catch, thank you!